src/runtime_src/core/tools/common/ReportElectrical.cpp

```diff
@@ -79,76 +33,20 @@ void
-    populate_sensor<qr::ddr_vpp_top_millivolts, qr::noop>(_pDevice, "ddr_vpp_top", "DDR Vpp Top")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v5v5_system_millivolts, qr::noop>(_pDevice, "5v5_system", "5.5 Volts System")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v1v2_vcc_top_millivolts, qr::noop>(_pDevice, "1v2_top", "Vcc 1.2 Volts Top")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v1v2_vcc_bottom_millivolts, qr::noop>(_pDevice, "vcc_1v2_btm", "Vcc 1.2 Volts Bottom")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v1v8_millivolts, qr::noop>(_pDevice, "1v8_top", "1.8 Volts Top")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v0v9_vcc_millivolts, qr::noop>(_pDevice, "0v9_vcc", "0.9 Volts Vcc")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v12v_sw_millivolts, qr::noop>(_pDevice, "12v_sw", "12 Volts SW")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::mgt_vtt_millivolts, qr::noop>(_pDevice, "mgt_vtt", "Mgt Vtt")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v3v3_vcc_millivolts, qr::noop>(_pDevice, "3v3_vcc", "3.3 Volts Vcc")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::hbm_1v2_millivolts, qr::noop>(_pDevice, "hbm_1v2", "1.2 Volts HBM")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v2v5_vpp_millivolts, qr::noop>(_pDevice, "vpp2v5", "Vpp 2.5 Volts")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v12_aux1_millivolts, qr::noop>(_pDevice, "12v_aux1", "12 Volts Aux1")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::noop, qr::vcc1v2_i_milliamps>(_pDevice, "vcc1v2_i", "Vcc 1.2 Volts i")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::noop, qr::v12_in_i_milliamps>(_pDevice, "v12_in_i", "V12 in i")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::noop, qr::v12_in_aux0_i_milliamps>(_pDevice, "v12_in_aux0_i", "V12 in Aux0 i")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::noop, qr::v12_in_aux1_i_milliamps>(_pDevice, "v12_in_aux1_i", "V12 in Aux1 i")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::vcc_aux_millivolts, qr::noop>(_pDevice, "vcc_aux", "Vcc Auxillary")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::vcc_aux_pmc_millivolts, qr::noop>(_pDevice, "vcc_aux_pmc", "Vcc Auxillary Pmc")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::vcc_ram_millivolts, qr::noop>(_pDevice, "vcc_ram", "Vcc Ram")));
-  sensor_array.push_back(std::make_pair("", 
-    populate_sensor<qr::v0v9_int_vcc_vcu_millivolts, qr::noop>(_pDevice, "0v9_vccint_vcu", "0.9 Volts Vcc Vcu")));
-  
-  pt.add_child("power_rails", sensor_array);
+  boost::property_tree::read_json(ss, pt);
+  ss.clear();

```

Comment:
Instead of re-using the stream, you could just "scope" the code block and declare the variable in the scope.   For example:
```
  boost::property_tree::ptree pt;
  {
    std::stringstream ss;
    ss << device.get_info<xrt::info::device::power_rails>();
    boost::property_tree::read_json(ss, pt);
  }

  boost::property_tree::ptree pt_power_consumption;
  {
    std::stringstream ss;
    ss << device.get_info<xrt::info::device::power_consumption>();
    boost::property_tree::read_json(ss, pt_power_consumption);
  }
```
Doing so indicates intent and keeps the code clean.

Reply:
done